src/core/services/theming/theming.spec.js

```diff
@@ -880,3 +880,89 @@ describe('md-themable directive', function() {
     expect(el.hasClass('md-default-theme')).toBe(true);
   }));
 });
+
+describe('$mdThemeProvider with custom styles that include nested rules', function() {
+  it('appends the custom styles taking into account nesting', function() {
+    module('material.core', function($mdThemingProvider) {
+      $mdThemingProvider.generateThemesOnDemand(false);
+      var styles =
+          "@media (min-width: 0) and (max-width: 700px) {\n\n"
+          + "  .md-THEME_NAME-theme .layout-row {\n"
+          + "    background-color: {{primary-500}};\n"
```

Comment:
The CSS rule's value should be in quotes (`background-color: "{{primary-500}}"`) to keep consistency with our documentation.